### PR TITLE
Add cvv cardTypeLength validator to allow card types to require a cer…

### PR DESCRIPTION
**Pull request by OzzieOrca**
…ain CVV length

- Set American Express to only allow CVV lengths of 4
- Remove unused cardType name field and rename displayName to name

Brought up by https://jira.cru.org/browse/EP-1756

### CODE REVIEW

**Comment by hlbraddock** [comment]
Is front end card validation simply a nicety? Normally, I'm an advocate for front end data validation. However, as it grows in complexity and is subject to evolving change over time, I wonder if it might not be worthwhile to simply rely on the the third party card services (TSEP/TransIT API) to handle validation?

File: src/credit-card/card-number/utils/card-types.ts [hunk 1]
```diff
@@ -1,35 +1,35 @@
...
-const cardTypes = [
+export const cardTypeConsts = [
   {
-    name: 'VISA',
-    displayName: 'Visa',
+    name: 'Visa',
     lengths: [13, 16],
-    prefixExpression: '4'
+    prefixExpression: '4',
+    expiryDateLengths: [3, 4]
```
Comment (hlbraddock) [inline src/credit-card/card-number/utils/card-types.ts]: Is this naming `cvv` length `expiryDateLengths`?

Comment (OzzieOrca) [inline src/credit-card/card-number/utils/card-types.ts]: Opps yep I'll rename that. Idk what I was thinking...

**Comment by OzzieOrca** [comment]
TSYS doesn't validate for this when doing the tokenization. Idk what happens on the backend when it gets processed. If you give it a amex card number and a 3 digit CVV, it happily tokenizes the credit card and returns it with no errors.

**Comment by hlbraddock** [comment]
I would be happy to have backed reject it, but that's one person's opinion. But not doing so means constantly updating the validation logic as card data rules evolve.

**Comment by hlbraddock** [comment]
I'll approve once you've renamed `cvvLengths`.

**Comment by OzzieOrca** [comment]
Updated. I think TSYS would need to do that backend validation. They might do it once the card is charged but by that point it is too late for the user to fix the CVV they typed. Our app backends will never see the CVV as far as I know.

**Comment by hlbraddock** [comment]
We're actually sending the cvv to the backend's, in fact. It's required to be sent along with the token.

**Comment by OzzieOrca** [comment]
Oh interesting. My assumption was that TSYS could use the token to look it up on their servers so it wasn't needed...

**Comment by hlbraddock** [comment]
That was everyone's reasonable assumption, including mine, until TSYS informed us that they are not allowed to store cvv's.

**Pull request merged**